mesonbuild/mparser.py

```diff
@@ -221,7 +221,7 @@ def lex(self, filename: str) -> T.Generator[Token, None, None]:
                     yield Token(tid, filename, curline_start, curline, col, bytespan, value)
                     break
             if not matched:
-                raise ParseException('lexer', self.getline(line_start), lineno, col)
+                raise ParseException(f'lexer: unrecognized token {self.code[loc]!r}', self.getline(line_start), lineno, col)
```

Comment:
We can do better: https://github.com/eli-schwartz/meson/commit/c4f18cb40f25632b4e31dc05b96aa0e607a4de0e

Unsure whether both commits should be used or not. Given a correct pointer, the error is now:

```
meson_options.txt:1:36: ERROR: lexer: unrecognized token ';'
option('foo', type: 'boolean', value; false, description: 'abc')
                                    ^
```

Just "lexer" is confusingly vague. Reporting the ";" twice might be duplicative. Maybe some different rewording should be used.

Reply:
I prefer being too explicit that not enough. First line gives what the error is, second (and third) line gives where it is. I think this is acceptable.